src/tools/sidctl/sidctl.c

```diff
@@ -48,6 +48,27 @@ struct args {
 	char **argv;
 };
 
+static int _usid_cmd_tree(struct args *args, output_format_t format, struct buffer *outbuf)
+{
+	struct buffer *         readbuf = NULL;
+	char **                 ptr;
+	size_t                  size;
+	struct usid_msg_header *msg;
+	int                       r;
+
+	if ((r = usid_req(LOG_PREFIX, USID_CMD_TREE, 0, NULL, NULL, &readbuf)) == 0) {
+		buffer_get_data(readbuf, (const void **) &msg, &size);
+		if (size < USID_MSG_HEADER_SIZE || msg->status & COMMAND_STATUS_FAILURE) {
+			buffer_destroy(readbuf);
+			return -1;
+		}
+		size -= USID_MSG_HEADER_SIZE;
+		buffer_add(outbuf, msg->data, size, &r);
```

Comment:
Hmm, we're reading almost the whole complete and formatted message here, just skipping the header. Then we're copying all the rest to the output buffer. It works fine, but would be nicer if we could avoid the extra copy.

Maybe we could define the output buffer as `BUFFER_TYPE_VECTOR` in which case we could add the pointer to the `msg->data` + `size` directly into the vector buffer as element. But right now, formatting is done on sidctl side for other command now and they use the output buffer for formatting so there the `BUFFER_TYPE_VECTOR` is not ideal. But this is caused by the fact that `tree` command is formatted on SID side while the rest is formatted on `sidctl` side at the moment.

So maybe we can keep this as it is for now (copying the `msg->data` to `outbuf` which is `BUFFER_TYPE_LINEAR`), later we can optimize and/or move the rest of commands to SID-side formatting.

Reply:
Ok.  I'll open an issue to fix.

Reply:
Fixed the warnings.  Sorry for that mistake.